File: ascenso-backoffice/src/services/actividades/NinoPadreService.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { NinoPadre, VincularPadreRequest } from '@/types/actividades'

class NinoPadreService {
  listar(destacamentoId: number, ninoId: number) {
    return apiClient.get<NinoPadre[]>(API_ENDPOINTS.ninoPadre.porNino(destacamentoId, ninoId))
  }

  vincular(destacamentoId: number, ninoId: number, request: VincularPadreRequest) {
    return apiClient.post<NinoPadre>(API_ENDPOINTS.ninoPadre.porNino(destacamentoId, ninoId), request)
  }

  desvincular(destacamentoId: number, ninoId: number, usuarioId: number) {
    return apiClient.delete<void>(API_ENDPOINTS.ninoPadre.porUsuario(destacamentoId, ninoId, usuarioId))
  }
}

export const ninoPadreService = new NinoPadreService()

File: ascenso-backend/src/main/java/com/ais/ascensobackend/actividades/application/services/interfaces/NinoPadreService.java
package com.ais.ascensobackend.actividades.application.services.interfaces;

import com.ais.ascensobackend.actividades.application.dtos.NinoPadreResumen;
import java.util.List;

public interface NinoPadreService {

    List<NinoPadreResumen> listar(Long destacamentoId, Long ninoId);

    NinoPadreResumen vincular(Long destacamentoId, Long ninoId, Long usuarioId);

    void desvincular(Long destacamentoId, Long ninoId, Long usuarioId);
}
